test/python/utest_pymasar/test_service_config.py

```diff
@@ -0,0 +1,165 @@
+            with self.assertRaises(Exception) as context:
+                saveServiceConfig(conn, servicename, masarconf, system=test_system, status=test_status,
+                                  configversion=test_version, configdesc=test_desc)
+            self.assertEqual('service config exists already.', context.exception.message)
+
+            with self.assertRaises(IndexError):  # TODO: Should this be a more specific error message?
+                saveServicePvGroup(conn, "bad config", [pvgname])
+            with self.assertRaises(Exception) as context:
+                res3 = saveServicePvGroup(conn, masarconf, ["bad pvgname"])  # this test prints a message to console
+            self.assertEqual("given pv group name (bad pvgname) does not exist.", context.exception.message)
+            res3 = saveServicePvGroup(conn, masarconf, [pvgname])
+            self.assertEqual([1], res3)
+            with self.assertRaises(Exception) as context:
+                pvlist = retrieveServiceConfigPVs(conn, masarconf, servicename="bad servicename")
+            pvlist = retrieveServiceConfigPVs(conn, masarconf, servicename=servicename)
+            self.assertEqual(context.exception.message, 'Given service (bad servicename) does not exist.')
+            self.assertEqual(pvs, pvlist)
+            results = retrieveServiceConfigs(conn, "bad servicename", masarconf)
+            self.assertEqual(1, len(results))  # IE: no data returned
+            self.assertEqual(6, len(results[0]))
+            self.assertEqual('config_idx', results[0][0])
+            self.assertEqual('config_name', results[0][1])
+            self.assertEqual('config_desc', results[0][2])
+            self.assertEqual('config_create_date', results[0][3])
+            self.assertEqual('config_version', results[0][4])
+            self.assertEqual('status', results[0][5])
+            results = retrieveServiceConfigs(conn, servicename, "bad conf")
+            self.assertEqual(1, len(results))  # IE: no data returned
+            self.assertEqual(6, len(results[0]))
+            self.assertEqual('config_idx', results[0][0])
+            self.assertEqual('config_name', results[0][1])
+            self.assertEqual('config_desc', results[0][2])
+            self.assertEqual('config_create_date', results[0][3])
+            self.assertEqual('config_version', results[0][4])
+            self.assertEqual('status', results[0][5])
+            results = retrieveServiceConfigs(conn, servicename, masarconf)
+            # Label tests
+            self.assertEqual(6, len(results[0]))
+            self.assertEqual(2, len(results))
+            self.assertEqual('config_idx', results[0][0])
+            self.assertEqual('config_name', results[0][1])
+            self.assertEqual('config_desc', results[0][2])
+            self.assertEqual('config_create_date', results[0][3])
+            self.assertEqual('config_version', results[0][4])
+            self.assertEqual('status', results[0][5])
```

Comment:
An easy way to avoid the repeated checks here would be to put the expected column names in a class variable, then use use ```assertListEqual(self.expect_cols, results[0])```.